lib/util.js

```diff
@@ -105,6 +109,12 @@ const numberRegExp = /^(0|[1-9][0-9]*)$/;
 
 const readableRegExps = {};
 
+// The actual maximum stack limit has to be divided by the number of function
+// calls that are maximal necessary for each depth level. That does not seem
+// to be sufficient though, so let us just divide it by 10 and use 1000 as upper
+// limit.
+const MAX_DEPTH_LIMIT = Math.min(Math.floor(getMaxStackLimit() / 10), 1000);
```

Comment:
This still seems too magical to me - no less magical than the `depth: 2` default at least, but that's simpler.